### DISCUSSION TITLE
xrdp-sesman listen on tcp6

### DISCUSSION THREAD

**Discussion by jonfon96** [neutrinolabs/xrdp#2857] [Q&A]
Hello All
I have Centos 7 install and I would like to use xrdp when I run it I have an issue with xrdp-sesman listen on tcp6 and xrdp listen on tcp. this is an issue when I want to connect to the Centos server via xrdp.  I looked on the site and I alter the port=3389 to port=tcp./3389 to only use tcp and not tcp6. I also disable the IPv6 on all the interface and restarted the server to confirm it took.  

I have looked on the web but nothing that comes out for changing the tcp6 to tcp for xrdp-sesman 

any help is welcomed. 

thank you

**Comment by matt335672** [comment]
The xrdp-sesman connection really shouldn't really be out on the network. It's not secure, and in the next major version of xrdp it's been removed and replaced with a Unix Domain Socket (#2207). 

Is there some feature of your local machine which makes a IPv4 connection unsuitable for this?

**Comment by jonfon96** [comment]
I do not understand the question but It does not allow me to rdp from a windows 10 PC to this server and the only issue, I think, is the tcp6 port xrdp-sesman is looking at.  If I am mistaken I would like to know how I can tell if this is or is not the issue.  
steps taken 
I have removed the ipv6 on all the interfaces on this device and rebooted to confirm its gone but I still see tcp6 ref in the netstat command.  

Some other post state I should remove it and recompile it from the install. not install from yum or dnf. 

let me know what you suggest to remove  tcp6 so it only used tcp. 

thank you and all and any information is welcomed.

**Comment by jonfon96** [comment]
Hello all I download the latest xrdp-0.9.23.1 and started the compiled if this program on Centos 7, I installed all the dependents noted in the readme and it seems to be flowing until I got to make and I get the following error. 

make[3]: Entering directory `/home/stsuser/software/xrdp/xrdp-0.9.23.1/sesman/chansrv'
sound.c: In function ‘process_pcm_message’:
sound.c:1121:21: error: ‘for’ loop initial declarations are only allowed in C99 mode
                     for (int i = 0; i < send_silence_times; i++)
                     ^
sound.c:1121:21: note: use option -std=c99 or -std=gnu99 to compile your code
make[3]: *** [sound.o] Error 1

I do not know what to do next since I am new to compiling this program.  I also run it as sudo but still the same issue .  I would like to understand what is the sound.c issue and why it stops there.  If someone can assist or what I need to modify or add to make this compile and complete. 

thank you

**Comment by jonfon96** [comment]
Hello All I looked around and found a good post to resolve the compile issue  https://stackoverflow.com/questions/14690142/gcc-error-for-loop-initial-declaration-used-outside-c99-mode

I located the section in sound.c in  /sesman/chansrv/ directory.  move the int outside of the for loop and it worked like a champ Data noted below.   
process_pcm_message(int id, int size, struct stream *s)
{
    static int sending_silence = 0;
    static int silence_start_time = 0;
    switch (id)
    {
        case 0:
            if ((g_client_does_fdk_aac || g_client_does_mp3lame) && sending_silence)
            {
                if ((g_time3() - silence_start_time) < 1000)
                {
                    /* do not send data within 1000mS after SNDC_CLOSE is sent. to avoid stutter */
                    return 0;
                }
                sending_silence = 0;
            }
            return sound_send_wave_data(s->p, size);
            break;
        case 1:
            if ((g_client_does_fdk_aac || g_client_does_mp3lame) && sending_silence == 0)
            {
                /* workaround for mstsc.exe. send silence data before send close */
                int send_silence_times = g_client_does_fdk_aac ? 4 : 2;  /* This value comes by trial and error */
                char *buf = (char *) g_malloc(g_bbuf_size, 0);
                if (buf != NULL)
                {
                    silence_start_time = g_time3();
                    sending_silence = 1;
                    for (int i = 0; i < send_silence_times; i++)                 <<-- error part
                    {
                        g_memset(buf, 0, g_bbuf_size);
                        sound_send_wave_data_chunk(buf, g_bbuf_size);
                    }
                    free(buf);
                    g_time_diff = 0;
                }
            }
            return sound_send_close();
            break;

Change the error part to 
int i;                                                  <<-- moved it outside the for loop. 
for ( i = 0; < send_silence_times; i++)                 <<-- new and update
                    {
                        g_memset(buf, 0, g_bbuf_size);
                        sound_send_wave_data_chunk(buf, g_bbuf_size);
                    }

It competed the compile but we will see if this resolved my tcp6 issue but I am not able to access the device.  

looking for more information to see why its not making connection.

**Comment by matt335672** [comment]
The connection made to xrdp is not the sesman connection.

There are two programs - xrdp and xrdp-sesman. xrdp-sesman listens locally for connections from xrdp. xrdp is the program which listens on the network.

This command will tell you how xrdp is listening on the network:-

```
ss -alp sport = 3389
```
